x/auth/tx/decoder.go

```diff
@@ -79,3 +90,28 @@ func DefaultJSONTxDecoder(cdc codec.ProtoCodecMarshaler) sdk.TxDecoder {
 		}, nil
 	}
 }
+
+// rejectNonADR027 rejects txBytes that do not adhere to ADR-027. This function
+// only checks that field numbers are in ascending order, as all other ADR-027
+// edge cases (e.g. TxRaw fields having default values) will not happen with
```

Comment:
the 5 rules were:
```
1. Fields must be serialized only once in ascending order
2. Extra fields or any extra data must not be added
3. [Default values](https://developers.google.com/protocol-buffers/docs/proto3#default)
   must be omitted
4. `repeated` fields of scalar numeric types must use
   [packed encoding](https://developers.google.com/protocol-buffers/docs/encoding#packed)
5. Varint encoding must not be longer than needed:
```

I'm checking (1) in this PR, (2) is checked by the unknown rejection. 3 and 4 won't happen (is that correct?).

But I think we actually need to check for 5 too.

Reply:
it's just 3 calls to `protowire.ConsumeField` checking that it's 1,2,3 with the right types

Reply:
also where would (5) be applicable?

Reply:
> also where would (5) be applicable?

For each field, we have `<1 byte that's tagNum | wireType> ++ <length prefix byte(s)> ++ <value bytes>`. I think the 2nd one can be any varint, i.e. longer than it actually needs to be.

Reply:
I see... well let's be as thorough as we need to be

Reply:
added varint length check